pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -478,14 +577,17 @@ func (o *EditOptions) visitToCreate(createVisitor resource.Visitor) error {
 func (o *EditOptions) visitAnnotation(annotationVisitor resource.Visitor) error {
 	// iterate through all items to apply annotations
 	err := annotationVisitor.Visit(func(info *resource.Info, incomingErr error) error {
-		// put configuration annotation in "updates"
-		if o.ApplyAnnotation {
-			if err := kubectl.CreateOrUpdateAnnotation(true, info, o.Encoder); err != nil {
+		if o.Record || cmdutil.ContainsChangeCause(info) {
+			if err := cmdutil.RecordChangeCause(info.Object, o.ChangeCause); err != nil {
 				return err
 			}
 		}
-		if o.Record || cmdutil.ContainsChangeCause(info) {
-			if err := cmdutil.RecordChangeCause(info.Object, o.ChangeCause); err != nil {
+		if o.EditMode == ApplyEditMode {
```

Comment:
I'd prefer keeping it as is (just using ApplyAnnotation) and making sure `ApplyAnnotation` is set as desired when in ApplyEditMode